lib/request_handler/json_api_document_parser.rb

```diff
@@ -27,9 +29,22 @@ def flattened_document
                                   detail: "Member 'data' is missing",
                                   source: { pointer: '/' } }]
       end
+      validate_json_api_data_members(resource)
       flatten_resource!(resource)
     end
 
+    def validate_json_api_data_members(data)
+      data.keys.each do |k|
+        next if VALID_DATA_MEMBERS.include?(k)
+
+        raise BodyParamsError, [{ code: 'INVALID_JSON_API',
```

Comment:
afaik the requesthandler does per default not verify if additional fields are passed in somewhere. Even though i like checking for extra things very much, I'm not sure if introducing this now is maybe not breaking something somewhere

Reply:
See PM